jina/peapods/compoundpod/__init__.py

```diff
@@ -382,3 +382,17 @@ def is_ready(self) -> bool:
             [p.is_ready.is_set() for p in self.peas]
             + [p.is_ready.is_set() for p in self.replica_list]
         )
+
+    def rolling_update(self):
+        """
+        Update all replicas of this compound pod.
+        """
+        for i in range(len(self.replica_list)):
+            replica = self.replica_list[i]
+            replica.close()
+            del replica
```

Comment:
deleting this replica has not real effect right.

You should make sure the context of this replica is exited.

At the end we went for the functional solution of replacing Pod? I like it, but wasnt it a little more complicated?

Reply:
> deleting this replica has not real effect right.
True, it gets garbage collected anyways.

> You should make sure the context of this replica is exited.
Yes, it is done by calling `replica.close()`

> At the end we went for the functional solution of replacing Pod? I like it, but wasn't it a little more complicated?
Yes, I inserted a timeout and stuff but it did not solve the issue that from time to time the test was hanging.
Therefore, I just block the flow until the rolling update is done.
I will create a follow-up ticket to have query while reloading.

Reply:
I would remove the delete.

As per query while reloading, the problem is not about havint this blocking or anything. 

I think the design problem is the need for Flow instance to query wrapping a client.

This should not be needed. We should separatr the client and the Flow instance